system-int-test/test/cmr/system_int_test/search/collection_shapefile_simplification_search_test.clj

```diff
@@ -0,0 +1,180 @@
+(ns cmr.system-int-test.search.collection-shapefile-simplification-search-test
+  (:require
+   [clojure.test :refer :all]
+   [clojure.java.io :as io]
+   [cheshire.core :as json]
+   [cmr.common.log :refer [debug]]
+   [cmr.common.mime-types :as mt]
+   [cmr.common.util :as util :refer [are3]]
+   [cmr.common-app.test.side-api :as side]
+   [cmr.search.services.parameters.converters.shapefile :as shapefile]
+   [cmr.search.middleware.shapefile :as shapefile-middleware]
+   [cmr.spatial.line-string :as l]
+   [cmr.spatial.mbr :as m]
+   [cmr.spatial.point :as p]
+   [cmr.spatial.polygon :as poly]
+   [cmr.system-int-test.data2.core :as d]
+   [cmr.system-int-test.data2.granule :as dg]
+   [cmr.system-int-test.data2.umm-spec-collection :as data-umm-c]
+   [cmr.system-int-test.utils.index-util :as index]
+   [cmr.system-int-test.utils.ingest-util :as ingest]
+   [cmr.system-int-test.utils.search-util :as search]
+   [cmr.umm.umm-spatial :as umm-s]
+   [cmr.system-int-test.data2.collection :as dc]))
+
+(use-fixtures :each (ingest/reset-fixture {"provguid1" "PROV1"}))
+
+(defn- polygon
+  "Creates a single ring polygon with the given ordinates. Points must be in counter clockwise
+  order."
+  [& ords]
+  (poly/polygon [(apply umm-s/ords->ring ords)]))
+
+(def formats
+  "Shapfile formats to be tested"
+  ;; ESRI is handled separately
+  {"GeoJSON" {:extension "geojson" :mime-type mt/geojson}
+   "KML" {:extension "kml" :mime-type mt/kml}})
+
+(defn make-coll
+  [coord-sys et & shapes]
+  (let [shapes (map (partial umm-s/set-coordinate-system coord-sys) shapes)]
+    (d/ingest "PROV1"
+              (dc/collection
+               {:entry-title et
+                :spatial-coverage (dc/spatial {:gsr coord-sys
+                                               :sr coord-sys
+                                               :geometries shapes})}))))
+
+(deftest collection-shapefile-simplification-search-test
+  (let [_ (side/eval-form `(shapefile/set-enable-shapefile-parameter-flag! true))
+        saved-shapefile-max-size (shapefile-middleware/max-shapefile-size)
+        _ (side/eval-form `(shapefile-middleware/set-max-shapefile-size! 2000000))
```

Comment:
This number doesn't match the max size in the documentation. Not sure if that's intentional.

Reply:
Yes, it's intentional. The sample files I have are over the 1MB limit (which is enforced before simplification can happen).